lib/ssl/src/tls_handshake_1_3.erl

```diff
@@ -650,13 +650,15 @@ do_start(#client_hello{cipher_suites = ClientCiphers,
         Cipher = Maybe(select_cipher_suite(HonorCipherOrder, ClientCiphers, ServerCiphers)),
         Groups = Maybe(select_common_groups(ServerGroups, ClientGroups)),
         Maybe(validate_client_key_share(ClientGroups, ClientShares)),
-        {PublicKeyAlgo, SignAlgo, SignHash, RSAKeySize} = get_certificate_params(Cert),
+        {PublicKeyAlgo, SignAlgo, SignHash, RSAKeySize, Curve} = get_certificate_params(Cert),
 
         %% Check if client supports signature algorithm of server certificate
+        %% TODO: We do validate the signature algorithm and signature hash but we could check
```

Comment:
You have a TODO comment here, I suppose we want to fix that?!

Reply:
Yes, I believe this can happen if a client connects to a faulty/malicious server that won't respect the signature_algorithms/signature_algorithms_cert extensions.